src/api/app/models/repository.rb

```diff
@@ -23,6 +23,8 @@ class Repository < ActiveRecord::Base
   validates :name, format: { with:    /\A[^_:\/\000-\037][^:\/\000-\037]+\Z/,
                              message: "Repository name must not start with '_' or contain any of these characters ':/'" }
 
+  # Name has to be unique among local repositories and remote_repositories of the associated db_project.
+  # Note that remote repositories also have to be unique among their remote project (remote_project_name)
```

Comment:
... have to be unique among their remote project (remote_project_name) and the project defining the remote.